src/rest/rest-api.php

```diff
@@ -24,7 +22,7 @@ public static function register_rest_routes() {
 				'id' => [
 					'validate_callback' => function( $param ) {
 						$post_id  = intval( $param );
-						$is_valid = 'publish' === get_post_status( $post_id );
+						$is_valid = self::is_post_readable( $post_id );
```

Comment:
Should update the comment above apply_filters to reflect the new logic for is_post_readable.

Reply:
Good catch! Updated the filter comment and corresponding README documentation.